Fix windows build error due to path too long

```
 C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppBuild.targets(312,5): error MSB3491: Could not write lines to file "x64\Release\vtkSlicerSkeletalRepresentationInitializerModuleLogicHierarchy\vtkSlice.273E9726.tlog\vtkSlicerSkeletalRepresentationInitializerModuleLogicHierarchy.lastbuildstate". The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters. [D:\D\P\SSALT-0-build\Slicer-build\E\SlicerSkeletalRepresentation\SkeletalRepresentationInitializer\Logic\vtkSlicerSkeletalRepresentationInitializerModuleLogicHierarchy.vcxproj] [D:\D\P\SSALT-0-build\slicersources-build\Slicer.vcxproj
```

See https://open.cdash.org/viewBuildError.php?buildid=5532315

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -166,7 +166,7 @@ list(APPEND Slicer_EXTENSION_SOURCE_DIRS ${${extension_name}_SOURCE_DIR})
 
 # Srep
 set(extension_name "SlicerSkeletalRepresentation")
-set(${extension_name}_SOURCE_DIR "${CMAKE_BINARY_DIR}/${extension_name}")
+set(${extension_name}_SOURCE_DIR "${CMAKE_BINARY_DIR}/SRep")
 FetchContent_Populate(${extension_name}
   SOURCE_DIR     ${${extension_name}_SOURCE_DIR}
   GIT_REPOSITORY ${EP_GIT_PROTOCOL}://github.com/KitwareMedical/SlicerSkeletalRepresentation

